taskfile/node_git_test.go

```diff
@@ -0,0 +1,75 @@
+package taskfile
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGitNode_ssh(t *testing.T) {
+	node, err := NewGitNode("git@github.com:foo/bar.git?ref=main//Taskfile.yml", "", false)
+	assert.NoError(t, err)
+	assert.Equal(t, "main", node.ref)
+	assert.Equal(t, "Taskfile.yml", node.path)
+	assert.Equal(t, "ssh://git@github.com/foo/bar.git?ref=main//Taskfile.yml", node.rawUrl)
+	assert.Equal(t, "ssh://git@github.com/foo/bar.git", node.URL.String())
+	entrypoint, err := node.ResolveEntrypoint("common.yml")
+	assert.NoError(t, err)
+	assert.Equal(t, "ssh://git@github.com/foo/bar.git?ref=main//common.yml", entrypoint)
+}
+
+func TestGitNode_sshWithDir(t *testing.T) {
+	node, err := NewGitNode("git@github.com:foo/bar.git?ref=main//directory/Taskfile.yml", "", false)
```

Comment:
> but using a compatible syntax would make it easier to transition to it in the future if we wanted to for any reason.

I did't thought about it, that would be a compelling reason to use that syntax.

I don't think this new change syntax is compatible...
```
git@github.com:foo/bar.git?ref=main//directory/Taskfile.yml
```
should be
```
git@github.com:foo/bar.git//directory/Taskfile.yml?ref=main
```

so `ref` is directly mapped to a git refeference, could be a branch or a tag for example.
id's say you'd need to split the url in `//` to get the git url at `0` and the filename at `1`

Reply:
hey @vmaerten thanks for all the work on this!

just want to add that with this syntax, the ref query string should be optional, because, it will only reference to a branch or tag, and without it it would be the default branch.

Reply:
Hey!
You're right! I'll work on this :)